File: JAVA_BACKEND/src/main/java/com/starads/dto/AplItemDto.java
package com.starads.dto;

import com.starads.generic.dto.GenericDTO;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AplItemDto implements GenericDTO<AplItemDto> {

	private Long id;

	private Long mplId;

	private String units;

	private Integer minVal;

	private Integer maxVal;

	private Integer thresHold;

	private String color;

	private Boolean displayOption;

	private String notification;

	private Integer severity;

}

File: NEST_JS_BACKEND/apps/starnavigationapi/src/app/modules/layers/dto/apl.item.dto..ts
import { ApiProperty } from "@nestjs/swagger";
import { Expose } from "class-transformer";
import { IsBoolean, IsNotEmpty, IsNumber, IsString } from "class-validator";
import { UserTypeEnum } from "../../../enum/user.type.enum";


/**
 * This is AplItems class
 * @ignore
 * Describes all the fields of APlItems for API 
 */
export class AplItemDto {
    @Expose()
    @ApiProperty({
        description: 'map id'
    })
    @IsNumber()
    mapId?: number;
    @IsString()
    @Expose()
    @ApiProperty({
        description: 'units'
    })
    @IsNotEmpty()
    units?: string;
    @Expose()
    @ApiProperty({
        description: 'min value'
    })
    @IsNumber()
    minVal?: number;
    @Expose()
    @ApiProperty({
        description: 'max value'
    })
    @IsNumber()
    maxVal?: number;
    @Expose()
    @ApiProperty({
        description: 'threshold'
    })
    @IsNumber()
    thresHold?: number;
    @IsString()
    @Expose()
    @ApiProperty({
        description: 'color'
    })
    @IsNotEmpty()
    color?: string;
    @IsBoolean()
    @Expose()
    @ApiProperty({
        description: 'Display option'
    })
    displayOption?: boolean;
    @IsString()
    @Expose()
    @ApiProperty({
        description: 'notification'
    })
    @IsNotEmpty()
    notification?: string;
    @Expose()
    @ApiProperty()
    @IsNotEmpty()
    severity?: number;
}